packages/@glimmer/runtime/lib/capabilities.ts

```diff
@@ -0,0 +1,31 @@
+import { ComponentCapabilities, Unique, Recast } from "@glimmer/interfaces";
+import { check, CheckNumber } from "@glimmer/debug";
+
+export type CapabilityFlags = Unique<"CapabilityFlag">;
+
+export const enum Capability {
+  DynamicLayout = 0b000001,
+  DynamicTag    = 0b000010,
+  PrepareArgs   = 0b000100,
+  CreateArgs    = 0b001000,
+  AttributeHook = 0b010000,
+  ElementHook   = 0b100000
+}
+
+/**
+ * Converts a ComponentCapabilities object into a 32-bit integer representation.
+ */
+export function capabilityFlagsFrom(capabilities: ComponentCapabilities): CapabilityFlags {
+  return (0 |
+    (capabilities.dynamicLayout ? Capability.DynamicLayout : 0) |
+    (capabilities.dynamicTag ? Capability.DynamicTag : 0) |
```

Comment:
You could use `capabilities.dynamicLayout | 0` to cast a boolean into `0` or `1`.

Reply:
TypeScript forbids bitwise operators on anything but numbers, enums or `any`.

Reply:
Yeah. Too bad. I really wish there was some way to disable this particular error in "boundary modules" like this but oh well.